extensions/binding/org.eclipse.smarthome.binding.mqtt.generic/src/main/java/org/eclipse/smarthome/binding/mqtt/generic/internal/convention/homie300/Property.java

```diff
@@ -146,30 +161,53 @@ public void attributesReceived() {
                 break;
         }
 
-        final ChannelState channelState = new ChannelState(stateTopic, commandTopic, channelUID, value);
+        final ChannelState channelState = new ChannelState(stateTopic, commandTopic, channelUID, value, callback);
+        channelState.setRetained(attributes.retained);
+        channelState.setReadOnly(!attributes.settable);
         this.channelState = channelState;
 
-        if (StringUtils.isBlank(stateTopic)) {
-            type = ChannelTypeBuilder.trigger(channelTypeUID, attributes.name).build();
-        } else {
+        final ChannelType type;
+        if (attributes.retained) {
             type = ChannelTypeBuilder.state(channelTypeUID, attributes.name, channelState.getItemType())
+                    .withConfigDescriptionURI(URI.create(MqttBindingConstants.CONFIG_HOMIE_CHANNEL))
                     .withStateDescription(value.createStateDescription(attributes.unit, !attributes.settable)).build();
+        } else {
+            type = ChannelTypeBuilder.trigger(channelTypeUID, attributes.name)
```

Comment:
Would it be possible to support button and rocker systemwise triggers based on the "$format" information provided by the homie attribute? The predefined systemwise channels are: system.rawbutton, system.button and system.rawrocker (https://www.eclipse.org/smarthome/documentation/development/bindings/thing-definition.html#channels).

- If the atribute "$datatype" is "enum" and "$format" is "PRESSED,RELEASED" -> system.rawbutton
- If the atribute "$datatype" is "enum" and "$format" is "SHORT_PRESSED,DOUBLE_PRESSED,LONG_PRESSED" -> system.button
- If the atribute "$datatype" is "enum" and "$format" is "DIR1_PRESSED,DIR1_RELEASED,DIR2_PRESSED,DIR2_RELEASED" -> system.rawrocker

System triggers can be used with default profiles and can be linked to items without the need of any rule. They very much improve the easy of use of the system for new users, so it would be ice on the cakes :+1: 

Thanks for your support @davidgraeff and looking forward to seeing this PR approved!

Reply:
I think the goal is to get this into the next OH release 2.4 including deprecating the existing mqtt binding. And that should happen around end of the week.
It would be contra productive to keep adding features at this point. Every feature needs another test case etc.
Make this a new Issue please and we thing about it for a later PR.

Reply:
Perfect, I will create an issue when this PR is merged :+1: